tests/Tests/Odesk/Phystrix/CommandTest.php

```diff
@@ -187,6 +187,70 @@ public function testRequestLogOff()
         $this->assertEmpty($this->requestLog->getExecutedCommands());
     }
 
+    /**
+     * Ensure command does not break when configured to log, though one hasn't been injected
+     *
+     * @dataProvider configBoolProvider
+     *
+     * @param bool $log_enabled  whether config is set to use request log
+     */
+    public function testRequestLogNotInjected($log_enabled) {
+
+        // Duplicate some of the class setup in order to bypass requestLog generation
+        $command = new CommandMock();
+        $command->setCommandMetricsFactory($this->commandMetricsFactory);
+        $command->setCircuitBreakerFactory($this->circuitBreakerFactory);
+        $command->setRequestCache(new RequestCache());
+
+        $command->setConfig(new \Zend\Config\Config(array(
+            'requestLog' => array(
+                'enabled' => $log_enabled,
+            ),
+        ), true));
+
+        $this->setUpCommonExpectations();
+
+        $this->assertEquals('run result', $this->command->execute());
+    }
+
+    /**
+     * Ensure command does not break when configured to cache, though cache hasn't been injected
+     *
+     * @dataProvider configBoolProvider
+     *
+     * @param bool $cache_enabled  whether config is set to use request log
+     */
+    public function testRequestCacheNotInjected($cache_enabled) {
+
+        // Duplicate some of the class setup in order to bypass requestLog generation
+        $command = new CommandMock();
+        $command->setCommandMetricsFactory($this->commandMetricsFactory);
+        $command->setCircuitBreakerFactory($this->circuitBreakerFactory);
+        $command->setRequestLog($this->requestLog);
+
+        $command->setConfig(new \Zend\Config\Config(array(
+            'requestCache' => array(
+                'enabled' => $cache_enabled,
+            ),
+        ), true));
+
+        $this->setUpCommonExpectations();
+
+        $this->assertEquals('run result', $this->command->execute());
+    }
+
+
+    /**
+     * @return array
+     */
+    public function configBoolProvider() {
+
+        return [
+            'config enabled'  => [ true ],
+            'config disabled' => [ false ],
```

Comment:
we have "php": ">=5.3.3" as requirement in composer.json. the brackets for arrays will require 5.4 at least